public/app/features/explore/TraceView/components/TraceTimelineViewer/SpanDetail/index.tsx

```diff
@@ -197,6 +218,46 @@ export default function SpanDetail(props: SpanDetailProps) {
       : []),
   ];
 
+  const [flameGraphFrame, setFlameGraphFrame] = React.useState<DataFrame>();
+
+  useEffect(() => {
+    if (config.featureToggles.traceToProfiles && request) {
+      let profilesDataSourceSettings: DataSourceInstanceSettings<DataSourceJsonData> | undefined;
+      if (traceToProfilesOptions?.datasourceUid) {
+        profilesDataSourceSettings = getDatasourceSrv().getInstanceSettings(traceToProfilesOptions.datasourceUid);
+      }
+      const hasPyroscopeProfile = span.tags.filter((tag) => tag.key === pyroscopeProfileTag).length > 0;
+
+      if (hasPyroscopeProfile && traceToProfilesOptions && profilesDataSourceSettings) {
+        const pyroRequest = {
+          ...request,
```

Comment:
Is the `request` reused to keep scoped vars and time range? I think it may contain requestId from Explore which may lead to unwanted query cancellation if two flamegraphs are shown

Reply:
@joey-grafana do we actually need the request? Not sure we need scoped vars as we create the query here anyway without any variables to interpolate and the time range should probably also go from the span (+- some buffer).

Reply:
Removed the request.